### PR TITLE
feat(SourceTree) - JS and React icons for source files

### PR DESCRIPTION
**Pull request by calebstdenis**
Associated Issue: #2545

### Summary of Changes
Source tree icons for JavaScript, CoffeeScript, React, and TypeScript.

### Test Plan
Tested with TypeScript, CoffeeScript, and React debugger examples, as seen in the screenshots below. In the case of the React example I had to go in and change the component file from ".js" to ".jsx" so it would show the React icon.

### Screenshots
image
------
image
------
image

^ It annoys me that the TS icon doesn't match the JS one, but it's all that was readily available. I'm thinking of making a separate issue for hunting down / making a matching SVG?

### CODE REVIEW

**Comment by calebstdenis** [comment]
Thinking of encapsulating `SourceTree` node logic into a `SourceTreeNode` component so I can snapshot test these guys. Thoughts?

**Comment by lukaszsobek** [comment]
For the icons you might find this interesting: https://github.com/devtools-html/debugger.html/issues/4141

**Comment by jasonLaster** [comment]
Thanks @lukaszsobek. I think these icons all look good :)

File: assets/images/coffeescript.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (wldcordeiro) [inline assets/images/coffeescript.svg]: Are we actually cool with putting the Mozilla license header on images of these library logos? I swear some may have trademarking or something.

**Comment by jasonLaster** [comment]
Looks like we need to handle the dark theme unselected case:

File: assets/images/coffeescript.svg (hunk 1 above)
Comment (calebstdenis) [reply assets/images/coffeescript.svg]: I'm no lawyer but afaik it's OK to use trademarks as long as you're not misleading people into thinking that you're associated with the company who owns it.

**Comment by calebstdenis** [comment]
@jasonLaster re dark theme: On it.

**Comment by calebstdenis** [comment]
@jasonLaster Fixed by setting the icons to the same color as the text

image
----
image

**Comment by jasonLaster** [comment]
looks good to me!

**Pull request merged**